tasks/insights-client-unregistration.yml

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Check if insights-client is installed
+  ansible.builtin.package_facts:
+    manager: auto
+
+- name: Call insights status
+  command: insights-client --status
+  failed_when: false
+  changed_when: false
+  register: __rhc_insights_status
+  when:
+    - '"insights-client" in ansible_facts.packages'
+
+- name: Unregister insights-client
+  command: insights-client --unregister
+  when:
+    - '"insights-client" in ansible_facts.packages'
+    - rhc_insights.state | d("present") == "absent"
+    - '"unregistered" not in __rhc_insights_status.stdout'
+    - '"NOT" not in __rhc_insights_status.stdout'
```

Comment:
Can you give me examples of the output when the system is registered, and when it is unregistered?

Reply:
It depends of the value of `legacy_upload` that is part of the configuration. The registration output  can be:
```
System is registered locally via .registered file. Registered at 2022-12-05T16:22:19.204489
Insights API confirms registration.
```
or
```
This host is registered
```

And  the unregistration:

```
System is NOT registered locally via .registered file. Unregistered at 2022-12-07T13:29:27.226725
Insights API says this machine is NOT registered.
System unregistered locally via .unregistered file
```
or

```
This host is unregistered
```